### ISSUE TITLE
cmake . -B build error

### ISSUE THREAD

**Issue by ZepSbosnia** [19reborn/NeuS2#27]
launching the command 
cmake . -B build
in the instant-ngp docker i get the following error and i cannot resolve it 

-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- The CUDA compiler identification is NVIDIA 10.2.89
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: /usr/local/cuda-10.2/bin/nvcc - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- No release type specified. Setting to 'Release'.
CMake Error at dependencies/neus2_tcnn/CMakeLists.txt:106 (list):
  list sub-command GET requires at least three arguments.

-- Targeting GPU architectures: 
-- Could NOT find Vulkan (missing: Vulkan_LIBRARY Vulkan_INCLUDE_DIR) 
CMake Warning at CMakeLists.txt:128 (message):
  Vulkan was not found.  Neural graphics primitives will still compile and
  run correctly, but DLSS will not be supported.

-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Using X11 for window creation
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found GLEW: /usr/include (found version "2.0.0") 
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- !!! Warning OptiX_INSTALL_DIR not set in environment. using default
-- OptiX_INSTALL_DIR value: /usr/local/NVIDIA-OptiX-SDK-7.3.0-linux64-x86_64
-- OptiX headers (optix.h and friends) not found.
CMake Warning at CMakeLists.txt:224 (message):
  OptiX was not found.  Neural graphics primitives will still compile and run
  correctly.  However, SDF training in 'raystab' and 'pathescape' modes will
  be significantly slower.

-- Found Python: /usr/local/bin/python3.7 (found suitable version "3.7.0", minimum required is "3.7") found components: Interpreter Development Development.Module Development.Embed 
-- pybind11 v2.7.1 
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Configuring incomplete, errors occurred!
See also "/volume/NeuS2/build/CMakeFiles/CMakeOutput.log".
See also "/volume/NeuS2/build/CMakeFiles/CMakeError.log".

**Comment by MikePelton** [comment]
Hi Zep - re the OptiX error, you need to install OptiX 7.6 in your docker (the newer OptiX 8 caused me some compilation errors although that may be fixed now) and then explicitly set the OptiX_INSTALL_DIR environment variable to point to it, e.g.
export OptiX_INSTALL_DIR=/home/blah/Downloads/NVIDIA-OptiX-SDK-7.6.0-linux64-x86_64/ 
Can't help with the Vulkan issue but again I'd imagine you'll need to install Vulkan and point to it via the Vulkan_LIBRARY and Vulkan_INCLUDE_DIR variables. Good luck!

**Comment by ZepSbosnia** [comment]
> 

somehow I've solved the problem but now I0m struggling here
I launch: 
cmake --build build --config RelWithDebInfo -j 
and get: 
[  3%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/common_device.cu.o
[  6%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/encoding.cu.o
[ 16%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/loss.cu.o
[ 19%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/cpp_api.cu.o
[ 19%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/cutlass_mlp.cu.o
[ 19%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/common.cu.o
[ 22%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/network.cu.o
[ 25%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/object.cu.o
[ 32%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/optimizer.cu.o
[ 32%] Building CUDA object dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/reduce_sum.cu.o
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:131: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/encoding.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/encoding.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:89: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/common_device.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/common_device.cu.o] Error 1
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:145: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/loss.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/loss.cu.o] Error 1
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:159: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/network.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/network.cu.o] Error 1
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:75: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/common.cu.o' failed
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
nvcc fatal   : Unsupported gpu architecture 'compute_sm_80'
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/common.cu.o] Error 1
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:103: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/cpp_api.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/cpp_api.cu.o] Error 1
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:173: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/object.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/object.cu.o] Error 1
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:117: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/cutlass_mlp.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/cutlass_mlp.cu.o] Error 1
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:187: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/optimizer.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/optimizer.cu.o] Error 1
dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/build.make:201: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/reduce_sum.cu.o' failed
make[2]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/reduce_sum.cu.o] Error 1
CMakeFiles/Makefile2:216: recipe for target 'dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/all' failed
make[1]: *** [dependencies/neus2_tcnn/src/CMakeFiles/tiny-cuda-nn.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by 19reborn** [comment]
Hi @ZepSbosnia , what's your cuda toolkit version? You may need to install cuda 11.0 or higher toolkit.

**Comment by ZepSbosnia** [comment]
i tried with many combination of cuda and torch but i'm not finding the right one. Can someone tell me a combination that works ??

> Hi @ZepSbosnia , what's your cuda toolkit version? You may need to install cuda 11.0 or higher toolkit.

**Comment by PannagaS** [comment]
Hi @ZepSbosnia, were you able to resolve this issue? I am also facing a similar issue now. I have cuda toolkit version 12.6 and my cuda version is 12.6 as well. I tried with cuda toolkit version 11.8, but it didn't work for me. Please let me know if you have any fix or anything that helped you resolve this cmake build issue. Thank you.

**Comment by MikePelton** [comment]
Hi all - can't give you a detailed response but this is coming from the fact your NVidia graphics cards are too new to support sm_80 / compute_80. Somewhere in the CMake make-file compute_80 is being specified for nvcc compiles - you just need to find out where and update it to something the Ada architecture cards will support (maybe sm_89?) (more explanation on the architectures [here](https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/)). There is no guarantee that what was built for sm_80 will still work with newer cards, but most of the time you'll get away with it. I built Neus2 a while ago (October 2023) on Ubuntu 22.04 with pytorch 2.0.1 and CUDA 11.8, and pytorch3D 0.7.4. One word of warning - I left myself a note at the time that the image data needed to be 32-bit PNG. Hope this helps - good luck!
